cookbooks/aws-parallelcluster-slurm/files/default/head_node_slurm/slurm/pcluster_slurm_config_generator.py

```diff
@@ -198,70 +198,108 @@ def _get_jinja_env(template_directory, realmemory_to_ec2memory_ratio):
     return env
 
 
-def _instance_type(compute_resource):
-    """Return the instance type in a compute resource.
+def _instance_types(compute_resource):
+    """Return the instance type list in a compute resource.
 
-    If the compute resource contains multiple instance types, it returns the first one.
+    If the compute resource is defined with single InstanceType, it returns a list of one element.
     """
     if compute_resource.get("InstanceTypeList"):
-        return compute_resource["InstanceTypeList"][0].get("InstanceType")
+        return [instance.get("InstanceType") for instance in compute_resource["InstanceTypeList"]]
     else:
-        return compute_resource["InstanceType"]
-
-
-def _gpu_count(instance_type):
-    """Return the number of GPUs for the instance."""
-    gpu_info = instance_types_data[instance_type].get("GpuInfo", None)
-
-    gpu_count = 0
-    if gpu_info:
-        for gpus in gpu_info.get("Gpus", []):
-            gpu_manufacturer = gpus.get("Manufacturer", "")
-            if gpu_manufacturer.upper() == "NVIDIA":
-                gpu_count += gpus.get("Count", 0)
-            else:
-                log.info(
-                    f"ParallelCluster currently does not offer native support for '{gpu_manufacturer}' GPUs. "
-                    "Please make sure to use a custom AMI with the appropriate drivers in order to leverage "
-                    "GPUs functionalities"
-                )
-
-    return gpu_count
-
-
-def _gpu_type(instance_type):
-    """Return name or type of the GPU for the instance."""
-    gpu_info = instance_types_data[instance_type].get("GpuInfo", None)
-    # Remove space and change to all lowercase for name
-    return "no_gpu_type" if not gpu_info else gpu_info.get("Gpus")[0].get("Name").replace(" ", "").lower()
+        return [compute_resource["InstanceType"]]
+
+
+def _get_min_gpu_count_and_type(instance_types) -> Tuple[int, str]:
```

Comment:
For readability, could we return a dictionary instead?
```
{"count": gpu_count, "type": gpu_type}
```

I've noticed the configuration is using the `first` and `last` jinja filters which makes it hard to follow without looking at the function return values again.

Reply:
done